pandas/core/frame.py

```diff
@@ -3418,48 +3417,48 @@ def sort_values(self, by, axis=0, ascending=True, inplace=False,
         if is_sequence(ascending) and len(by) != len(ascending):
             raise ValueError('Length of ascending (%d) != length of by (%d)' %
                              (len(ascending), len(by)))
+        if not isinstance(ascending, (tuple, list)):
+            ascending = [ascending] * len(by)
+        ascending = ascending[::-1]
+
+        new_data = None
+
+        from pandas.core.sorting import nargsort
+
         if len(by) > 1:
-            from pandas.core.sorting import lexsort_indexer
+            kind = 'mergesort'
 
-            def trans(v):
-                if needs_i8_conversion(v):
-                    return v.view('i8')
-                return v
-
-            keys = []
-            for x in by:
-                k = self.xs(x, axis=other_axis).values
-                if k.ndim == 2:
-                    raise ValueError('Cannot sort by duplicate column %s' %
-                                     str(x))
-                keys.append(trans(k))
-            indexer = lexsort_indexer(keys, orders=ascending,
-                                      na_position=na_position)
-            indexer = _ensure_platform_int(indexer)
-        else:
-            from pandas.core.sorting import nargsort
+        for i, by_step in enumerate(by[::-1]):
```

Comment:
This one I had to change to new_data = self because later I use an xs indexer. There might be nicer way of writing the code (without xs) but I need to dive more into internals of Pandas, this is my first attempt to help with development.